java_sdk_example/src/main/java/aws/sample/paymentcryptography/hmac/HMACService.java

```diff
@@ -1,86 +1,103 @@
+                .keyAlgorithm(KeyAlgorithm.TDES_2_KEY)
+                .keyClass(KeyClass.SYMMETRIC_KEY)
+                .keyUsage(KeyUsage.TR31_M3_ISO_9797_3_MAC_KEY)
+                .keyModesOfUse(modes)
+                .build();
+                
+        CreateKeyRequest request = CreateKeyRequest.builder()
+        .keyAttributes(attributes)
+        .enabled(true)
+        .exportable(true)
+        .build();
 
-        Key key = ControlPlaneUtils.getControlPlaneClient().createKey(request).getKey();
-        ControlPlaneUtils.upsertAlias(hmacKeyAlias.getAliasName(), key.getKeyArn());
-        return hmacKeyAlias.getAliasName();
+        Key key = ControlPlaneUtils.getControlPlaneClient().createKey(request).key();
+        ControlPlaneUtils.upsertAlias(hmacKeyAlias.aliasName(), key.keyArn());
+        return hmacKeyAlias.aliasName();
     }
 
-    public String generateMac(String text) {
+    public String generateMac(String text) throws InterruptedException, ExecutionException {
         String hmacKeyArn = getHMACKey();
-        GenerateMacResult macGenerateResult = generateMac(hmacKeyArn,text);
-        return macGenerateResult.getMac();
+        GenerateMacResponse macGenerateResponse = generateMac(hmacKeyArn,text);
+        return macGenerateResponse.mac();
     }
 
-    public GenerateMacResult generateMac(String hmacKeyArn, String text) {
-        MacAttributes macAttributes = new MacAttributes()
-                .withAlgorithm(MacAlgorithm.ISO9797_ALGORITHM3);
+    public GenerateMacResponse generateMac(String hmacKeyArn, String text) {
+        MacAttributes macAttributes = MacAttributes
+                .builder()
+                .algorithm(MacAlgorithm.ISO9797_ALGORITHM3)
+                .build();
+
         Logger.getGlobal().info("HMACService:generateMac Attempting to generate HMAC thru AWS Cryptography Service for text " + text);
-        GenerateMacRequest generateMacRequest = new GenerateMacRequest()
-                .withKeyIdentifier(hmacKeyArn)
-                .withMessageData(Hex.encodeHexString(text.getBytes()))
-                .withGenerationAttributes(macAttributes);
-        GenerateMacResult macGenerateResult = DataPlaneUtils.getDataPlaneClient().generateMac(generateMacRequest);
-        Logger.getGlobal().info("HMACService:generateMac HMAC generation successfult for " + text + ". HMAC is " + macGenerateResult.getMac());
-        return macGenerateResult;
+        GenerateMacRequest generateMacRequest = GenerateMacRequest
+                .builder()
+                .keyIdentifier(hmacKeyArn)
+                .messageData(Hex.encodeHexString(text.getBytes()))
+                .generationAttributes(macAttributes)
+                .build();
+
+        GenerateMacResponse macGenerateResponse = DataPlaneUtils.getDataPlaneClient().generateMac(generateMacRequest);
+        Logger.getGlobal().info("HMACService:generateMac HMAC generation successfult for " + text + ". HMAC is " + macGenerateResponse.mac());
```

Comment:
I never noticed this but this should be called MAC Service not HMAC service since HMAC isn't what's being used.

Reply:
done